src/node.rs

```diff
@@ -1832,16 +1832,130 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EthNamespaceT for
 
     fn get_block_transaction_count_by_number(
         &self,
-        _block_number: zksync_types::api::BlockNumber,
+        block_number: zksync_types::api::BlockNumber,
     ) -> jsonrpc_core::BoxFuture<jsonrpc_core::Result<Option<U256>>> {
-        not_implemented("get_block_transaction_count_by_number")
+        let inner = Arc::clone(&self.inner);
+
+        Box::pin(async move {
+            let maybe_result = {
+                let reader = match inner.read() {
+                    Ok(r) => r,
+                    Err(_) => return Err(into_jsrpc_error(Web3Error::InternalError)),
+                };
+                match block_number {
```

Comment:
suggestion: Feels like we can reduce the amount of code here by calculating the block_hash first:

```rust
let block_hash = match block_number {
    zksync_types::api::BlockNumber::Latest
    | zksync_types::api::BlockNumber::Pending
    | zksync_types::api::BlockNumber::Finalized
    | zksync_types::api::BlockNumber::Committed => reader
        .block_hashes
        .get(&reader.current_miniblock),
    zksync_types::api::BlockNumber::Number(ask_number) => reader
        .block_hashes
        .get(&ask_number.as_u64()),
    zksync_types::api::BlockNumber::Earliest => reader
        .block_hashes
        .get(&0),
};

reader
    .blocks
    .get(block_hash)
    .map(|block| U256::from(block.transactions.len()))
    .or_else(|| {
        reader
            .fork_storage
            .inner
            .read()
            .expect("failed reading fork storage")
            .fork
            .as_ref()
            .and_then(|fork| {
                fork.fork_source
                    .get_block_transaction_count_by_number(block_number)
                    .ok()
                    .flatten()
            })
    })
```